metadata-ingestion/src/datahub/ingestion/extractor/mce_extractor.py

```diff
@@ -1,27 +1,44 @@
+import time as t
 from typing import Iterable, Union
 
 from datahub.ingestion.api import RecordEnvelope
 from datahub.ingestion.api.common import PipelineContext
 from datahub.ingestion.api.source import Extractor, WorkUnit
 from datahub.ingestion.api.workunit import MetadataWorkUnit, UsageStatsWorkUnit
-from datahub.metadata.com.linkedin.pegasus2avro.mxe import MetadataChangeEvent
+from datahub.metadata.com.linkedin.pegasus2avro.mxe import (
+    MetadataChangeEvent,
+    SystemMetadata,
+)
 from datahub.metadata.schema_classes import UsageAggregationClass
 
 
+def current_milli_time():
+    return round(t.time() * 1000)
+
+
 class WorkUnitRecordExtractor(Extractor):
     """An extractor that simply returns the data inside workunits back as records."""
 
+    ctx: PipelineContext
+
     def configure(self, config_dict: dict, ctx: PipelineContext) -> None:
-        pass
+        self.ctx = ctx
 
     def get_records(
         self, workunit: WorkUnit
     ) -> Iterable[RecordEnvelope[Union[MetadataChangeEvent, UsageAggregationClass]]]:
         if isinstance(workunit, MetadataWorkUnit):
+
+            system_metadata = SystemMetadata(
+                lastObserved=current_milli_time(), runId=self.ctx.run_id
+            )
```

Comment:
note: the extractor runs before the transformers, and so any MCEs created by a transformer won't get the system metadata :(

This isn't an immediate issue since our current transformers only modify MCEs and don't create any, but it will definitely come up.

Reply:
Ok- good tip. Any suggestions on how to best circumvent this?

Reply:
We'll need to split the extractor up into (1) extracting RecordEnvelopes from work units and (2) running validation + checks. (2) can probably be implemented as a transformer that always runs